src/components/app/data/utils.js

```diff
@@ -367,17 +368,40 @@ export const transformSubsidyRequest = ({
   notifications: [], // required prop by CourseSection
 });
 
+export const assignmentTypeStates = ({ state }) => ({
+  isAcceptedAssignment: state === ASSIGNMENT_TYPES.ACCEPTED,
+  isAllocatedAssignment: state === ASSIGNMENT_TYPES.ALLOCATED,
+  isCanceledAssignment: state === ASSIGNMENT_TYPES.CANCELED,
+  isExpiredAssignment: state === ASSIGNMENT_TYPES.EXPIRED,
+  isErroredAssignment: state === ASSIGNMENT_TYPES.ERRORED,
+  isExpiringAssignment: state === ASSIGNMENT_TYPES.EXPIRING,
+});
+
 export const transformLearnerContentAssignment = (learnerContentAssignment, enterpriseSlug) => {
-  const isCanceledAssignment = learnerContentAssignment.state === ASSIGNMENT_TYPES.CANCELED;
-  const isExpiredAssignment = learnerContentAssignment.state === ASSIGNMENT_TYPES.EXPIRED;
-  const { date: assignmentEnrollByDeadline } = learnerContentAssignment.earliestPossibleExpiration;
+  const {
+    contentKey,
+    parentContentKey,
+    isAssignedCourseRun,
+    state,
+    earliestPossibleExpiration,
+  } = learnerContentAssignment;
+  const {
+    isExpiredAssignment,
+    isCanceledAssignment,
+  } = assignmentTypeStates({ state });
+  const { date: assignmentEnrollByDeadline } = earliestPossibleExpiration;
+
+  let courseKey = contentKey;
```

Comment:
[nit] might be helpful to include a comment on why the modification of `courseKey` and `courseRunId` is happening (i.e., something along the lines of "to remain backwards compatible with assignments for top-level courses" or similar).